toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt

```diff
@@ -55,27 +60,58 @@ public class MapViewScope(internal val mapView: MapView)
  * @since 200.5.0
  */
 @Composable
-public fun MapViewScope.Callout(location: Point,
-                                modifier: Modifier = Modifier,
-                                content: @Composable BoxScope.() -> Unit) {
+public fun MapViewScope.Callout(
+    location: Point,
+    modifier: Modifier = Modifier,
+    content: @Composable BoxScope.() -> Unit
+) {
+    val isMapViewReady = remember { mutableStateOf(false) }
+    // We don't want to start drawing the Callout until the MapView is ready. We only collect
+    // the drawStatus till the first time MapView is done drawing. the transformWhile operator
+    // will stop collecting when isMapViewReady.value becomes false.
+    LaunchedEffect(location) {
+        mapView.drawStatus.transformWhile { drawStatus ->
+            emit(drawStatus)
+            !isMapViewReady.value
+        }.collect {
+            if (it == DrawStatus.Completed) {
+                isMapViewReady.value = true
+            }
+        }
+    }
+
+    if (!isMapViewReady.value) {
+        return
+    }
 
-    if (mapView.map?.loadStatus?.collectAsState()?.value == LoadStatus.Loaded) {
-        val calloutScreenCoordinate: ScreenCoordinate = mapView.locationToScreen(location)
-        Box(
-            modifier = Modifier
-                .graphicsLayer {
-                    translationX = calloutScreenCoordinate.x.toFloat()
-                    translationY = calloutScreenCoordinate.y.toFloat()
-                }
-                .wrapContentSize()
-                .background(Color.White)
-                .border(
-                    border = BorderStroke(2.dp, Color.LightGray),
-                    shape = MaterialTheme.shapes.medium
-                )
-        )
-        {
-            this.content()
+    // Convert the given location to a screen coordinate
+    var calloutScreenCoordinate: ScreenCoordinate by remember {
```

Comment:
possibly this could key on `location` then it doesn't need to be assigned in the LaunchedEffect.
```suggestion
    var calloutScreenCoordinate: ScreenCoordinate by remember(location) {
```

Reply:
I think this is not the key but the initial value for the calloutScreenCoordinate. we are using the changes to the state of calloutScreenCoordinate to trigger recomposition.